mlx_graphs/utils/convert.py

```diff
@@ -30,19 +34,95 @@ def to_networkx(
         OutEdgeView([(0, 0), (0, 1), (1, 0), (1, 2), (2, 1), (2, 3), (3, 2)])
 
     """
-    import networkx as nx
 
     G = nx.DiGraph()
 
+    if data.graph_features is not None:
+        G.graph["graph_features"] = data.graph_features.tolist()
+
+    if data.graph_labels is not None:
+        G.graph["graph_labels"] = data.graph_labels.tolist()
+
     if data.num_nodes is None:
         return G
 
-    G.add_nodes_from(range(data.num_nodes))
+    # G.add_nodes_from(range(data.num_nodes))
+
+    node_features_present = data.node_features is not None
+    node_labels_present = data.node_labels is not None
+
+    node_attrs = {}
+    for i in range(data.num_nodes):
+        if node_features_present:
+            node_attrs["features"] = data.node_features[i].tolist()
+        if node_labels_present:
+            node_attrs["label"] = data.node_labels[i].item()
+        G.add_node(i, **node_attrs)
 
-    for v, w in data.edge_index.T.tolist():
+    edge_features_present = data.edge_features is not None
+    edge_attrs = {}
+    for i, (v, w) in enumerate(data.edge_index.T.tolist()):
         if remove_self_loops and v == w:
             continue
-
-        G.add_edge(v, w)
+        if edge_features_present:
+            edge_attrs["features"] = data.edge_features[i].tolist()
+        G.add_edge(v, w, **edge_attrs)
 
     return G
+
+
+def from_networkx(data: Union[nx.Graph]) -> GraphData:
```

Comment:
`Union[nx.Graph]` -> `Union[nx.Graph, nx.DiGraph]`?

Reply:
nx.Digraph inherits from graph so I guess we this can cover both cases
https://networkx.org/documentation/networkx-1.10/_modules/networkx/classes/digraph.html#DiGraph

Reply:
Ok, then I guess the `Union` is not needed

Reply:
Can I keep it `Any` here and then try to import `networkx` that way the user will not have to install networkx for every function in the future

Reply:
No, should we need to, we'll make networkx a dependency, but I'd rather not use `Any`